lib/IRGen/GenCall.cpp

```diff
@@ -5030,16 +5030,23 @@ Callee irgen::getSwiftFunctionPointerCallee(
   return Callee(std::move(calleeInfo), fn, dataPtr);
 }
 
-Callee irgen::getCFunctionPointerCallee(IRGenFunction &IGF,
-                                        llvm::Value *fnPtr,
+Callee irgen::getCFunctionPointerCallee(IRGenFunction &IGF, llvm::Value *fnPtr,
                                         CalleeInfo &&calleeInfo) {
   auto sig = emitCastOfFunctionPointer(IGF, fnPtr, calleeInfo.OrigFnType);
-  auto authInfo =
-    PointerAuthInfo::forFunctionPointer(IGF.IGM, calleeInfo.OrigFnType);
 
-  auto fn = FunctionPointer::createSigned(FunctionPointer::Kind::Function,
-                                          fnPtr, authInfo, sig);
+  if (!IGF.getSILModule()
+           .getOptions()
+           .EnableImportPtrauthFieldFunctionPointers) {
+    auto sig = emitCastOfFunctionPointer(IGF, fnPtr, calleeInfo.OrigFnType);
+    auto authInfo =
+        PointerAuthInfo::forFunctionPointer(IGF.IGM, calleeInfo.OrigFnType);
 
+    auto fn = FunctionPointer::createSigned(FunctionPointer::Kind::Function,
+                                            fnPtr, authInfo, sig);
+    return Callee(std::move(calleeInfo), fn);
+  }
+  auto fn = FunctionPointer::createUnsigned(FunctionPointer::Kind::Function,
+                                            fnPtr, sig);
```

Comment:
We should not create an unsigned pointer here.

If the flag is not used (or you have a function pointer without the ptrauth qualifier) we use the default pointer signing for c function pointers: they are all signed the same.

For example:

```
% cat fnptr.h
#ifndef fnptr_h
#define fnptr_h

#include <stddef.h>

typedef struct {
    int x;
    void (* _Nonnull fnptr)(size_t);
} Container;

#endif
% xcrun -sdk iphoneos swiftc -import-objc-header fnptr.h -target arm64e-apple-ios13 fnptr.swift -emit-ir 2>&1 | less
...
  call void %23(i64 5) [ "ptrauth"(i32 0, i64 0) ]
  ret void
}
```

Reply:
I realize the above comment is not totally correct because I misread the code. 

You use unsigned  pointers if `EnableImportPtrauthFieldFunctionPointers` is true. (I read the code the opposite round)

That is still not what we want. Because if the ptrauth qualifier is missing we want to use whatever signed scheme applies.